### ISSUE TITLE
Duplicate variable declarations wrongly removed

### ISSUE THREAD

**Issue by adammockor** [rollup/rollup#716]
If I import [foundation](https://github.com/zurb/foundation-sites/blob/develop/dist/foundation.js) as

```
import {} from './node_modules/foundation-sites/dist/foundation.js';
```
only last _createClass helper definition stays, others gets removed. Problem is that every definition of _createClass helper is in their own iffe scope, so I am getting 'undefined' on _createClass helpers calls.

If I import foundation plugins one by one as 

```
import {} from './node_modules/foundation-sites/dist/plugins/foundation.core.js';
import {} from './node_modules/foundation-sites/dist/plugins/foundation.abide.js';
import {} from './node_modules/foundation-sites/dist/plugins/foundation.orbit.js';
```
it works as expected, _createClass helpers gets namespacing and they are not removed.

rollup 0.31.0, no plugins

**Comment by eventualbuddha** [comment]
[Minimal reproduction](http://rollupjs.org/#%7B%22options%22%3A%7B%22format%22%3A%22cjs%22%2C%22moduleName%22%3A%22myBundle%22%2C%22globals%22%3A%7B%7D%7D%2C%22modules%22%3A%5B%7B%22name%22%3A%22main.js%22%2C%22code%22%3A%22var%20a%20%3D%201%3B%5Cn%5Cnconsole.log(a)%3B%5Cn%5Cnvar%20a%20%3D%201%3B%5Cn%5Cnconsole.log(a)%3B%22%7D%5D%7D):

```js
// main.js
var a = 1;

console.log(a);

var a = 1;

console.log(a);


// output.js
'use strict';

console.log(a);

var a = 1;

console.log(a);
```

**Comment by Rich-Harris** [comment]
This is proving surprisingly finicky to fix – I feel a bit of a refactoring coming on – but how does this general solution sound: in a situation like this, we simply drop the `var` from all but the first declaration, and treat subsequent ones as assignments instead (this is essentially what happens if you run the code).

```js
var a = 1;

console.log(a);

a = 1;

console.log(a);
```

At the same time we should throw an error if a duplicate declaration is encountered and either the first or the second isn't a function or `var` declaration (since it's a syntax error to redeclare a `let`/`const`/`class`).

**Comment by Rich-Harris** [comment]
*Finally* fixed in 0.35.0

**Comment by snuggs** [comment]
HOLY F*&K I WAS WORKING ON THIS AND GOT STUCK ! THANK YOU @Rich-Harris! This was breaking with the Acorn update. been working on async/await stuff.
